File: src/main/java/com/ddd/trainings/idea/domain/IdeaNameVO.java
package com.ddd.trainings.idea.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.jmolecules.ddd.annotation.ValueObject;

@ValueObject
@AllArgsConstructor
public class IdeaNameVO {

  @Getter
  private final String nameValue;

  public static IdeaNameVO from(String name) {
    //TODO: Implement validation of idea name
    return new IdeaNameVO(name);
  }

  public String[] getKeywords() {
    return nameValue.split(" ");
  }
}

File: src/Ddd/Trainings/Idea/Domain/IdeaNameVO.cs
namespace Ddd.Trainings.Idea.Domain
{
    public class IdeaNameVO(string nameValue)
  {
    public string NameValue { get; } = nameValue;

    public static IdeaNameVO From(string name)
        {
            //TODO: Implement validation of idea name
            return new IdeaNameVO(name);
        }

        public string[] Keywords => NameValue.Split(' ');
    }
} 